From c7187da2c5eb447f89b5668d6be85749fd2e84e5 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Tue, 6 Feb 2018 19:55:31 +0900
Subject: [PATCH 573/909] arm64: dts: r8a77990: Add DEVFREQ support for GSX

commit d6294bfe8ad10a4afc7a11ae1d778146a8c64f1a from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch adds opp table for GSX to use devfreq.

NOTE: Frequency scaling is not supported.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
[van.do.xw: fix frequency and voltage of GSX opp.]
Signed-off-by: Van Do <van.do.xw@rvc.renesas.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
[takeshi.kihara.df: fix W=1 dtc unit_address_vs_reg warnings]
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a77990.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a77990.dtsi b/arch/arm64/boot/dts/renesas/r8a77990.dtsi
index 8cf1f915a090..5132b06e2687 100644
--- a/arch/arm64/boot/dts/renesas/r8a77990.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77990.dtsi
@@ -131,6 +131,19 @@
 		clock-frequency = <0>;
 	};
 
+	gsx_opp_table: gsx_opp_table {
+		compatible = "operating-points-v2";
+
+		opp-300000000 {
+			opp-hz = /bits/ 64 <300000000>;
+			opp-microvolt = <820000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <820000>;
+		};
+	};
+
 	/* MSIOF reference clock - to be overridden by boards that provide it */
 	msiof_ref_clk: msiof-ref-clock {
 		compatible = "fixed-clock";
@@ -1508,6 +1521,7 @@
 			reg = <0 0xfd000000 0 0x40000>;
 			interrupts = <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 112>;
+			operating-points-v2 = <&gsx_opp_table>;
 			power-domains = <&sysc R8A77990_PD_3DG_B>;
 			resets = <&cpg 112>;
 		};
-- 
2.17.1

